### PR TITLE
More client features

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

* Fill out most parameters of start/execute workflow
* Build search attribute collection
* Switched durations from being millisecond ints to second floats
* Expose RPC metadata and timeout to high-level calls
* More explicit generated code for gRPC both Ruby and Rust side
* Support for proper gRPC status details in RPC errors
* Workflow already started error
* Cloud ops protos
* Added RBS signature files
* Added CI check for protos
* Added a couple of tests (many more to come)
* Docs fixes/improvements

### CODE REVIEW

**Review by dandavison — APPROVED** [review]
I'm approving this. Obviously it's a lot of code, mostly scaffolding, and hard to review in a thorough sense. I've looked through, and learned a few things, and it all looks great and carefully done. I don't think there's a need at this early stage of the project to block merging this on careful review by others.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,10 +71,19 @@ jobs:
...
+      - name: Install bundle
+        working-directory: ./temporalio
+        run: bundle install
+
+      - name: Check generated protos
+        if: ${{ matrix.checkTarget }}
+        working-directory: ./temporalio
+        run: |
+          bundle exec rake proto:generate
+          [[ -z $(git status --porcelain lib/temporalio/api) ]] || (git diff lib/temporalio/api; echo "Protos changed"; exit 1)
```
[the remark this answers is not in the document]
Comment (cretz) [reply .github/workflows/ci.yml:83]: This was taken from https://github.com/temporalio/sdk-python/blob/4aef4bfda19e56c7e7abac36b69e98f60861468c/.github/workflows/ci.yml#L95 so I wonder if we want to fix there too?

[the remark this answers is not in the document]
Comment (dandavison) [reply .github/workflows/ci.yml:83]: Yep sounds good. Not urgent of course; could do it now or just do it next time one of us touches the CI yaml.

File: temporalio/Rakefile [hunk 2]
```diff
@@ -34,57 +36,273 @@ require 'yard'
...
+
+    # Generate Rust code
+    def generate_rust_match_arm(file:, qualified_service_name:, service_enum:, trait:)
+      # Do service lookup
+      desc = Google::Protobuf::DescriptorPool.generated_pool.lookup(qualified_service_name)
+      file.puts <<~TEXT
+        #{service_enum} => match call.rpc.as_str() {
+      TEXT
+
+      # desc.sort_by { |a, b| a.name <=> b.name }.each do |method|
```
[the remark this answers is not in the document]
Comment (cretz) [reply temporalio/Rakefile]: Fixed

**Pull request merged**